### PR TITLE
[O2B-1253] Set DPL detectors order in RCT views

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- Set DPL detectors order in RCT views 

Notable changes for developers:
- NA

Changes made to the database:
- NA

### CODE REVIEW

File: lib/public/services/detectors/detectorsNames.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+export const DETECTOR_NAME = {
```
Comment (martinboulais) [inline lib/public/services/detectors/detectorsNames.js:14]: What is this? Maybe VirtualDetectorsNames or anything but not just detectors names

**Review by martinboulais — DISMISSED** [review]

File: lib/public/services/detectors/detectorsNames.js (hunk 1 above)
Comment (xsalonx) [reply lib/public/services/detectors/detectorsNames.js]: renamed

File: lib/public/services/detectors/detectorsNames.js
Comment (graduta) [inline lib/public/services/detectors/detectorsNames.js:1]: Is it `lib/public/domain/enums` not a better place for this?
Moreover, please add some docs on what a virtual detector is and its usage

File: lib/public/services/detectors/dplDetectorsProvider.js [hunk 2]
```diff
@@ -30,7 +29,15 @@ export class DplDetectorsProvider extends Observable {
...
             .apply((remoteDetectors) => remoteDetectors.apply({
-                Success: (detectors) => detectors.filter(({ name }) => name && name !== TST_DETECTOR_NAME),
+                Success: (detectors) => {
+                    const filteredDetectors = detectors.filter(({ name }) => name && name.toUpperCase() !== VIRTUAL_DETECTOR_NAME.TST);
+                    const gloIndex = filteredDetectors.findIndex(({ name }) => name.toUpperCase() === VIRTUAL_DETECTOR_NAME.GLO);
+                    if (gloIndex !== -1) {
+                        const [gloDetector] = filteredDetectors.splice(gloIndex, 1);
+                        return [gloDetector, ...filteredDetectors.sort(({ name: nameA }, { name: nameB }) => nameA < nameB)];
+                    }
+                    return filteredDetectors.sort();
```
Comment (graduta) [inline lib/public/services/detectors/dplDetectorsProvider.js:39]: You are currently iterating 5 times over these arrays with the `filter`, `findIndex`, `splice`, 2x`sort`
I would recommend you improve it with 2 iterations:
- 1 `for` in which you build a new array which does not contain TST or GLO. In the same for you identify the index of `GLO`
- 1 sort of the newly created array + return of GLO if present

**Review by graduta — CHANGES_REQUESTED** [review]

File: lib/public/services/detectors/detectorsNames.js
Comment (xsalonx) [reply lib/public/services/detectors/detectorsNames.js:1]: done

File: lib/public/services/detectors/dplDetectorsProvider.js (hunk 2 above)
Comment (xsalonx) [reply lib/public/services/detectors/dplDetectorsProvider.js]: refactored

**Review by graduta — APPROVED** [review]

**Pull request merged**